nixos/modules/virtualisation/containers.nix

```diff
@@ -22,37 +22,224 @@ let
 
   # The container's init script, a small wrapper around the regular
   # NixOS stage-2 init script.
-  containerInit = pkgs.writeScript "container-init"
-    ''
-      #! ${pkgs.stdenv.shell} -e
+  containerInit = (cfg:
+    let
+      renderExtraVeth = (name: cfg:
+        ''
+        echo "Bringing ${name} up"
+        ip link set dev ${name} up
+        ${if cfg . "localAddress" or null != null then
```

Comment:
`if condition then "foo" else ""` is more neatly expressed as `optionalString condition "foo"`.